### ISSUE TITLE
[a11y]: unable to change the level of heading in SidePanel and its children headings properly

### ISSUE THREAD

**Issue by tshimizu-ibm** [carbon-design-system/ibm-products#7274]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

Windows

### Package version

v2.63.0

### React version

v18.3.1

### Automated testing tool and ruleset

IBM Accessibility Requirements 1.3.1 Info and Relationships

### Description

While I'm trying to make our product to meet IBM a11y requirement 1.3.1 https://www.ibm.com/able/requirements/requirements/?version=v7_3#1_3_1, I found the title of `SidePanel` is marked as `<h2>` hardcoded in https://github.com/carbon-design-system/ibm-products/blob/778347ee7a4bb1bf08b385177707ee9fc409d324/packages/ibm-products/src/components/SidePanel/SidePanel.tsx#L696-L711
so we cannot align the level of panel inside level 2 or higher content semantically and also cannot mark headings properly in children components of `SidePanel`.

It should be Carbon's `<Heading>` component instead of `<h2>` tag and wrap the entire panel with Carbon's `<Section>` to dynamically align the level including given children components inside properly.

### Reproduction/example

https://codesandbox.io/p/devbox/cool-smoke-74c48y

### Steps to reproduce

1. Open the example app in Chrome
2. Click "Open side panel" button in the app preview
3. Open Chrome Dev Tools, or inspect "Incident management" title of the side panel
4. Find `SidePanel`'s heading is marked as `<h2>` even though `<SidePanel>` is set inside the section of "Third heading" `<h3>`, and headings of `<ChildrenContent />` "Subtitle" are marked as `<h4>`. In this case "Incident management" and "Subtitle"s are expected to be marked as `<h4>` and `<h5>` respectively.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by elycheea** [comment]
@tshimizu-ibm Thanks for opening this! We’ve been making the updates to use `Section` and `Heading` in several components already. We’ll add the side panel to the list as well. 🙏
